app/models/client.rb

```diff
@@ -9,25 +9,30 @@ class Client < ApplicationRecord
   validates :name, :code, presence: true
   validate :code_is_valid
 
-  def balance
-    begin
-      response = Faraday.get("http://localhost:4000/api/v1/balance/#{self.code_numbers}")
-      body = JSON.parse(response.body)
-      return body["balance"]
-    rescue => exception
-      return 0
-    end
-  end
+  after_create :create_wallet
 
   def code_numbers
     code.split('-').join.split('.').join.split('/').join
   end
 
   private
 
+  def create_wallet
+    begin
+      params = { client_wallet: { email:, registered_number: code } }
+      response = Faraday.post('http://localhost:4000/api/v1/client_wallets', params)
+
+      if response.status.digits.last == 2 || response.body.include?('em uso')
+        self.update(has_wallet: true)
+      end
+    end
+  end
+
   def code_is_valid
     return unless code
 
-    errors.add :code, 'inválido' if !CNPJ.valid?(code) && !CPF.valid?(code)
+    return if CNPJ.valid?(code) || CPF.valid?(code) && code.include?('.')
+
+    errors.add :code, 'inválido'
```

Comment:
Essa mensagem poderia ser um pouco mais completa, algo como:
```suggestion
    errors.add :code, 'CPF/CNPJ inválido'
```